### PR TITLE
Vijayan graph

### PR DESCRIPTION
**Pull request by vijayankrishnan**
Site bandwidth component changes added.

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -22,11 +22,15 @@
     "@angular/platform-browser": "^4.0.0",
     "@angular/platform-browser-dynamic": "^4.0.0",
     "@angular/router": "^4.0.0",
+    "@swimlane/ngx-charts": "^7.0.0",
     "bootstrap-sass": "^3.3.7",
     "core-js": "^2.4.1",
+    "d3": "^4.11.0",
     "font-awesome": "^4.7.0",
     "jquery": "^3.2.1",
+    "moment": "^2.19.2",
```
Comment (happiestRaviThapa) [inline package.json]: do we really need MomentJS ?

File: src/app/components/dashboard/float-site-bandwidth-info/float-site-bandwidth-info.component.html [hunk 2]
```diff
@@ -12,44 +12,66 @@
...
-							<i class="fa fa-calendar fa-lg" aria-hidden="true"></i> 10 Oct, 2017
-							<span>{{ 'SiteBandwidthInfo.bandwidthTo' | translate}}</span>
-							<i class="fa fa-calendar fa-lg" aria-hidden="true"></i> 15 Oct, 2017
+						<div class="select-date-range datetime-popup col-sm-8">
+							<div class="from-date-lable">{{ 'SiteBandwidthInfo.bandwidthFrom' | translate}}</div>
+							<div class="from-date-icon">
+								<input class="form-control" type="hidden" [(ngModel)]="fromDate" />
+								<div class="btn" (click)="onTogglePicker('fromDate')">
+									<i class="fa fa-calendar" aria-hidden="true"></i>
+									{{fromDate | date: 'MMM d, y'}}
```
Comment (happiestRaviThapa) [inline src/app/components/dashboard/float-site-bandwidth-info/float-site-bandwidth-info.component.html]: add 'fromDate in en.json'

File: src/app/components/dashboard/float-site-bandwidth-info/float-site-bandwidth-info.component.html [hunk 3]
```diff
@@ -12,44 +12,66 @@
...
+								<datetime-popup class="datetime-popup" [value]="fromDate" [maxDate]="maxDate" (valueChange)="onFromDateChange($event)" [(showPopup)]="fromDateShowPicker"
+								 [showDate]="showDate" [showTime]="showTime" [closeButton]="closeButton" class="datetimePickerPopup"></datetime-popup>
+
+							</div>
+							<div class="to-date-lable">{{ 'SiteBandwidthInfo.bandwidthTo' | translate}}</div>
+							<div class="to-date-icon">
+								<input class="form-control" type="hidden" [(ngModel)]="toDate" />
+								<div class="btn" (click)="onTogglePicker('toDate')">
+									<i class="fa fa-calendar" aria-hidden="true"></i>
+									{{toDate | date: 'MMM d, y'}}
```
Comment (happiestRaviThapa) [inline src/app/components/dashboard/float-site-bandwidth-info/float-site-bandwidth-info.component.html]: en.json

File: src/app/components/dashboard/float-site-bandwidth-info/float-site-bandwidth-info.component.html [hunk 4]
```diff
@@ -12,44 +12,66 @@
...
+								</div>
+								<datetime-popup [value]="toDate" [minDate]="minDate" [maxDate]="maxDate" (valueChange)="onToDateChange($event)" [(showPopup)]="toDateShowPicker"
+								 [showDate]="showDate" [showTime]="showTime" [closeButton]="closeButton" class="datetimePickerPopup"></datetime-popup>
+
+							</div>
 						</div>
 						<div class="fullscreen-config col-sm-2">
 							<div class="fullscreen">
-								<i class="fa fa-window-maximize fa-lg" aria-hidden="true"></i>
+								<!-- <i class="fa fa-window-maximize fa-lg" aria-hidden="true"></i> -->
```
Comment (happiestRaviThapa) [inline src/app/components/dashboard/float-site-bandwidth-info/float-site-bandwidth-info.component.html]: add comment, saying 'required for later use'

File: src/app/components/dashboard/ngx-charts/line-chart-demo.ts [hunk 5]
```diff
@@ -0,0 +1,65 @@
...
+    'Monotone Y', 'Natural', 'Step', 'Step After', 'Step Before'
+];
+colorScheme: object = {};
+  constructor( private _bandwidth: FloatSiteBandwidthInfoComponent ) {
+    this.colorScheme = _bandwidth.colorScheme;
+      Object.assign(this, {multi}) 
+  }
+  
+  onSelect(event) {
+    console.log(event);
```
Comment (happiestRaviThapa) [inline src/app/components/dashboard/ngx-charts/line-chart-demo.ts]: something meaningful name & log

File: src/app/components/dashboard/ngx-charts/site-bandwidth-chart.html [hunk 6]
```diff
@@ -0,0 +1,23 @@
+<div class="row">
+  <div class="col-md-6">
+    <div style="display: block;">
```
Comment (happiestRaviThapa) [inline src/app/components/dashboard/ngx-charts/site-bandwidth-chart.html]: no inline css

File: package.json (hunk 1 above)
Comment (vijayankrishnan) [reply package.json]: removed

File: src/app/components/dashboard/float-site-bandwidth-info/float-site-bandwidth-info.component.html (hunk 2 above)
Comment (vijayankrishnan) [reply src/app/components/dashboard/float-site-bandwidth-info/float-site-bandwidth-info.component.html]: Its a model and loads dynamically on date change

File: src/app/components/dashboard/float-site-bandwidth-info/float-site-bandwidth-info.component.html (hunk 3 above)
Comment (vijayankrishnan) [reply src/app/components/dashboard/float-site-bandwidth-info/float-site-bandwidth-info.component.html]: Its a model and loads dynamically on date change

File: src/app/components/dashboard/float-site-bandwidth-info/float-site-bandwidth-info.component.html (hunk 4 above)
Comment (vijayankrishnan) [reply src/app/components/dashboard/float-site-bandwidth-info/float-site-bandwidth-info.component.html]: comments added

File: src/app/components/dashboard/ngx-charts/line-chart-demo.ts (hunk 5 above)
Comment (vijayankrishnan) [reply src/app/components/dashboard/ngx-charts/line-chart-demo.ts]: not required and removed

File: src/app/components/dashboard/ngx-charts/site-bandwidth-chart.html (hunk 6 above)
Comment (vijayankrishnan) [reply src/app/components/dashboard/ngx-charts/site-bandwidth-chart.html]: not required and removed

**Comment by vijayankrishnan** [comment]
changes added as per code review

**Pull request merged**